library/src/amd_detail/rocblaslt/src/kernels/compile_code_object.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# ########################################################################
+# Copyright (C) 2022-2024 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell cop-
+# ies of the Software, and to permit persons to whom the Software is furnished
+# to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+# PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+# CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# ########################################################################
+sources=$1
+archs=$2
+dest=$3
+rocm_path="${ROCM_PATH:-/opt/rocm}"
+hipcc_path="${rocm_path}/bin/hipcc"
```

Comment:
hipcc has been deprecated. Please switch to amdclang

Reply:
Unfortunately, here we cannot simply switch to amdclang since the `--genco` option is not supported by amdclang, there's a special route to handle `--genco` in hipcc, refer to [LLVM repo.](https://github.com/ROCm/llvm-project/blob/fa1653be3807a0b59d6f8730ee26cef7f4ba7ae8/amd/hipcc/src/hipBin_amd.h#L536)

Reply:
Switched to amdclang, thanks for YangWen's help!